pkg/service/FluxService/helper.go

```diff
@@ -0,0 +1,363 @@
+	if err != nil {
+		return nil
+	}
+	metadata, _, err := unstructured.NestedMap(rowObject, k8sCommonBean.K8sClusterResourceMetadataKey)
+	if err != nil {
+		return nil
+	}
+	if !shouldProcessApp(FluxAppType, metadata) {
+		return nil
+	}
+	rowCells, _, err := unstructured.NestedSlice(rowDataMap, k8sCommonBean.K8sClusterResourceCellKey)
+	if err != nil {
+		return nil
+	}
+	name, syncStatus, healthStatus := extractValuesFromRowCells(rowCells, columnDefinitions)
+
+	namespace, found, err := unstructured.NestedString(metadata, "namespace") // Replace "namespace" with actual key
+	if err != nil || !found {
+		return nil
+	}
+	return &FluxApplicationDto{
+		Name:         name,
+		HealthStatus: healthStatus,
+		SyncStatus:   syncStatus,
+		EnvironmentDetails: &EnvironmentDetail{
+			ClusterId:   clusterId,
+			ClusterName: clusterName,
+			Namespace:   namespace,
+		},
+		FluxAppDeploymentType: FluxAppType,
+	}
+}
+func GetApplicationListDtos(resources unstructured.UnstructuredList, clusterName string, clusterId int, FluxAppType FluxAppType) []*FluxApplicationDto {
+	manifestObj := resources.Object
+	fluxAppDetailArray := make([]*FluxApplicationDto, 0)
+
+	columnDefinitions, found, err := unstructured.NestedSlice(manifestObj, k8sCommonBean.K8sClusterResourceColumnDefinitionKey)
+	if err != nil || !found {
+		return fluxAppDetailArray
+	}
+
+	columnDefinitionMap := extractColumnDefinitions(columnDefinitions)
+
+	rowsData, found, err := unstructured.NestedSlice(manifestObj, k8sCommonBean.K8sClusterResourceRowsKey)
+	if err != nil || !found {
+		return fluxAppDetailArray
+	}
+
+	for _, rowData := range rowsData {
+		rowDataMap, ok := rowData.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		appDto := createFluxApplicationDto(rowDataMap, columnDefinitionMap, clusterId, clusterName, FluxAppType)
+		if appDto != nil {
+			fluxAppDetailArray = append(fluxAppDetailArray, appDto)
+		}
+	}
+	return fluxAppDetailArray
+}
+func GetFluxAppDetailDto(appDetail *FluxApplicationDto) *client.FluxApplication {
+	return &client.FluxApplication{
+		Name:                  appDetail.Name,
+		HealthStatus:          appDetail.HealthStatus,
+		SyncStatus:            appDetail.SyncStatus,
+		FluxAppDeploymentType: string(appDetail.FluxAppDeploymentType),
+		EnvironmentDetail: &client.EnvironmentDetails{
+			ClusterName: appDetail.EnvironmentDetails.ClusterName,
+			ClusterId:   int32(appDetail.EnvironmentDetails.ClusterId),
+			Namespace:   appDetail.EnvironmentDetails.Namespace,
+		},
+	}
+}
+func getFluxSpecKubeConfig(obj map[string]interface{}) (bool, error) {
+	spec, found, err := unstructured.NestedMap(obj, "spec")
+	if err != nil {
+		return false, err
+	}
+	if !found {
+		return false, nil
+	}
+	_, found, err = unstructured.NestedFieldCopy(spec, "kubeConfig")
+	if err != nil {
+		return false, err
+	}
+	if !found {
+		return false, nil
+	}
+	return true, nil
+}
+func parseObjMetadata(s string) (FluxKsResourceDetail, error) {
```

Comment:
also put comment that this func logic belongs to flux cd, and don't change the internal implementation of this func, before changing look into flux cd code